### PR TITLE
jobs/bump-lockfile: fix builds when missing architectures

### PR DESCRIPTION
**Pull request by dustymabe**
Also includes a commits to rework the structure to loop more.

```
commit 9f0fdfde7bfed8cbf447954236af83e7484a7c33
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Fri Jul 22 13:59:44 2022 -0400

    jobs/bump-lockfile: more deduplication
    
    Put the fetach build and tests into a loop as well. Almost all
    of the code is the same except for the testiso bits which I've
    added conditional logic to.

commit c309e52620939ae59018f663f79e17999a09bc92
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Fri Jul 22 13:18:20 2022 -0400

    jobs/bump-lockfile: use new --find-build-for-arch buildfetch option
    
    Right now the latest build in a stream could have failed the secondary
    architectures runs and have missing arches. This causes the `cosa build`
    to fail because it tries to read the meta data for the latest build
    but none exists for this arch. Let's have buildfetch traverse the
    build history to find the latest build and fetch that build's info
    for each respective architecture.

commit ac2ca16a821adaca89b59119d59396c2e5eb0dfb
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Fri Jul 22 13:15:15 2022 -0400

    jobs/bump-lockfile: store session in archinfo; add loops
    
    This reworks bump-lockfile to loop over the archinfo structure
    and dedupe some code. It also stores the session information
    in the archinfo data structure.

```

### CODE REVIEW

**Comment by dustymabe** [comment]
added one more commit: 

```
commit b11ce5c251b0d598e1df3e7b56ed3d1b2982bed7
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Fri Jul 22 14:09:10 2022 -0400

    jobs/bump-lockfile: support ALLOW_KOLA_UPGRADE_FAILURE
    
    If we get in a bind and need to work around an issue let's have
    a knob for it like we do the other jobs.

```

**Comment by dustymabe** [comment]
this PR requires: https://github.com/coreos/coreos-assembler/pull/3006

File: jobs/bump-lockfile.Jenkinsfile [hunk 1]
```diff
@@ -397,14 +392,12 @@ try { lock(resource: "bump-${params.STREAM}") { timeout(time: 120, unit: 'MINUTE
 
     // Destroy the remote sessions. We don't need them anymore
     stage("Destroy Remotes") {
-        parallel aarch64: {
-            pipeutils.withExistingCosaRemoteSession(arch: "aarch64",
-                                                    session: session_aarch64) {
-                shwrap("cosa remote-session destroy")
+        for (arch in archinfo.keySet()) {
+            if (arch == "x86_64") {
```
Comment (cgwalters) [inline jobs/bump-lockfile.Jenkinsfile:306]: Optional followup: Perhaps worth factoring out `def remoteArches = archinfo.keySet().filter(|| arch != "x86_64")` or so?  (may not be valid groovy syntax)

**Review by cgwalters — APPROVED** [review]
LGTM as is!

**Pull request merged**